native/objcexport-header-generator/src/org/jetbrains/kotlin/backend/konan/objcexport/objcTypes.kt

```diff
@@ -106,9 +106,21 @@ data class ObjCBlockPointerType(
         append(attrsAndName)
         append(")(")
         if (parameterTypes.isEmpty()) append("void")
-        parameterTypes.joinTo(this) { it.render() }
+        var index = 0
+        parameterTypes.joinTo(this) { it.render(getParameterName(index++)) }
         append(')')
     })
+
+    private fun getParameterName(index: Int): String {
+        // When using suspend functions, additional block parameters are generated,
+        // while the original list of parameter names remains empty.
+        if (originParameterNames.isEmpty()) return ""
+
+        assert(originParameterNames.size == parameterTypes.size) {
+            "ObjC Block parameter type size(${parameterTypes.size}) should equal to originParameterNames size(${originParameterNames.size})."
+        }
+        return originParameterNames[index]
```

Comment:
Here things are getting dangerous: some Kotlin parameter names can be invalid for Objective-C block type parameter names. Try `int`, for example.
Some kind of sanitizing and mangling is necessary here.

Also, this feature can't be turned off. So, a single tricky name somewhere among the dependencies would break the whole compilation, causing a critical regression. Please add a way to turn off the whole change with a [binary option](https://github.com/JetBrains/kotlin/blob/8d0f5b64f005dbb0e78514c9aebc455a3cba1a54/kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/BinaryOptions.kt#L15).